File: #1-25 (C, Java, Python)/Java/Euler25.java
//Euler25 ~ 1000-digit Fibonacci number
import java.util.concurrent.TimeUnit;
import java.math.BigInteger;

public class Euler25
{
	public static void main(String[] args)
	{
		long startTime = System.nanoTime();
		
		BigInteger thousandDigits = new BigInteger("10").pow(999);
		BigInteger[] fibbonaci    = new BigInteger[3];
		fibbonaci[0] = BigInteger.ONE;
		fibbonaci[2] = BigInteger.ONE;
		
		int digits = 2;
		int i = 0;
		while(fibbonaci[i].compareTo(thousandDigits) < 0)
		{
			i = (i+1)%3;
			fibbonaci[i] = fibbonaci[(i+1)%3].add(fibbonaci[(i+2)%3]);
			digits++;
		}
		
		long endTime = System.nanoTime();
		long timeElapsed = endTime - startTime;
		System.out.println("Solution: " + digits + "\n");
		System.out.println("Program completed in " + timeElapsed/1000000000 + " seconds.");
	}
}

File: #1-25 (C, Java, Python)/Python/Euler25.py
#Euler25 ~ 1000-digit Fibonacci number
import timeit
start = timeit.default_timer()

fibonnaci = [1,1]
i = 2
while 1:
    next = fibonnaci[i-1] + fibonnaci[i-2]
    
    if(len(str(next)) == 1000):
        break
    else:
        fibonnaci.append(next)
    
    i += 1

solution = i + 1

stop = timeit.default_timer()
print("Solution: ",solution,"\n")
print("Program completed in","{0:.3f}".format(stop - start), "seconds.")